lib/shelf_io.dart

```diff
@@ -96,26 +109,28 @@ Future handleRequest(HttpRequest request, Handler handler) async {
 
   // TODO(nweiz): abstract out hijack handling to make it easier to implement an
   // adapter.
-  Response response;
+  Response? response;
   try {
     response = await handler(shelfRequest);
   } on HijackException catch (error, stackTrace) {
     // A HijackException should bypass the response-writing logic entirely.
     if (!shelfRequest.canHijack) return;
 
     // If the request wasn't hijacked, we shouldn't be seeing this exception.
-    response = _logError(shelfRequest,
-        "Caught HijackException, but the request wasn't hijacked.", stackTrace);
+    response = _logError(
+      shelfRequest,
+      "Caught HijackException, but the request wasn't hijacked.",
+      stackTrace,
+    );
   } catch (error, stackTrace) {
-    response =
-        _logError(shelfRequest, 'Error thrown by handler.\n$error', stackTrace);
+    response = _logError(
+      shelfRequest,
+      'Error thrown by handler.\n$error',
+      stackTrace,
+    );
   }
 
-  if (response == null) {
-    await _writeResponse(_logError(shelfRequest, 'null response from handler.'),
```

Comment:
Was this overly defensive (dead) code before? I generally don't expect so see this type of change in migrations to no longer handle null where it was previously handled.

Reply:
this was dead, I think. There are no code paths where `response` could be null. Finally caught it.

...at least I think!

Reply:
Running tests with `pub run --no-sound-null-safety` shows a failure. It looks related to this.